src/main/kotlin/com/github/seliba/devcordbot/command/impl/CommandClientImpl.kt

```diff
@@ -158,6 +177,24 @@ class CommandClientImpl(
         return findCommand(Arguments(input.trim().split(delimiter), raw = input), commandAssociations)
     }
 
+    private fun stripPrefix(message: Message): String? {
+        val rawInput = message.contentRaw
+
+        when (message.channelType) {
+            ChannelType.TEXT -> {
+                val prefixLength = resolvePrefix(message.guild, rawInput) ?: return null
+                return rawInput.substring(prefixLength).trim()
+            }
+            ChannelType.PRIVATE -> {
+                val prefix = prefix.find(rawInput) ?: return null
+                return rawInput.substring(prefix.range.last + 1).trim()
+            }
+            else -> {
+                return null
+            }
+        }
+    }
+
     private fun resolvePrefix(guild: Guild, content: String): Int? {
```

Comment:
```suggestion
private fun resolvePrefix(guild: Guild?, content: String): Int? {
        val mention = guild?.selfMember?.asNickedMention()

        val prefix = prefix.find(content)
        return when {
            mention?.let { content.startsWith(it) } == true -> mention.length
            prefix != null -> prefix.range.last + 1
            else -> null
        }
    }
```

This should work for both, and yes nickedMention always works DK why